app/test-utils/runTestSuite.js

```diff
@@ -0,0 +1,58 @@
+import nock from 'nock';
+import { ManifestProvider } from '../manifestProvider';
+import { runCommonComponentsTests } from './testFunctions/commonComponentsTests';
+import { runQuestionTests } from './testFunctions/questionTests';
+
+const apiLocalhost = 'http://localhost:8080';
+
+export const runTestSuite = async ({
+  data,
+  sectionApiUrl,
+  sectionId,
+  clientUrl,
+  parentSectionApiUrl,
+}) => {
+  const mocks = (responseStatus, responseBody) => {
+    nock(apiLocalhost)
+      .get(sectionApiUrl)
+      .reply(responseStatus, responseBody);
+  };
+
+  const pageSetup = async ({
+    t,
+    responseStatus = 200,
+    responseBody = {},
+  }) => {
+    mocks(responseStatus, responseBody);
+    await t.navigateTo(clientUrl);
+  };
+
+  const sectionManifest = new ManifestProvider().getSectionManifest(sectionId);
+
+  fixture(`Show ${sectionManifest.title} page`)
+    .afterEach(async (t) => {
+      const isDone = nock.isDone();
+      if (!isDone) {
+        nock.cleanAll();
+      }
+      await t.expect(isDone).ok('Not all nock interceptors were used!');
+    });
+
+  runCommonComponentsTests({
+    pageSetup,
+    sectionManifest,
+    sectionApiUrl,
+    data,
+    parentSectionApiUrl,
+    apiLocalhost,
+  });
+
+  runQuestionTests({
+    pageSetup,
+    sectionManifest,
+    sectionApiUrl,
+    data,
+    sectionId,
+    apiLocalhost,
+  });
```

Comment:
How about we move the runErrorTests into here? Don't think it fits into the runQuestionTests suite. What do you think?

Reply:
I think it needs to go in here because of the recursion on contact details as each has an errorResponse

Reply:
😞 now the recursion has make me sad. Ok happy to leave as is